### PR TITLE
docs: update moderator handbook

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #XXXXX

Updates the handbook to rewrite the Discord section for the new chat server, removing Discord entirely.

### CODE REVIEW

**Review by raisedadead — COMMENTED** [review]
Holding this up for a bit to coincide with the announcement.

**Comment by nhcarrigan** [comment]
Sounds good.

File: docs/moderator-handbook.md
Comment (marsonya) [inline docs/moderator-handbook.md:234]:
```suggestion
In all other situations - even situations where the code of conduct is violated - moderators should not delete the messages as they are important historic records. When you do delete a message, make sure you take a screenshot of it first! The screenshot can be logged in the #mod-log channel.
```

File: docs/moderator-handbook.md
Comment (marsonya) [inline docs/moderator-handbook.md:227]:
```suggestion
4. **Deleting messages** Moderators can delete messages on our chat server. They should only exercise this ability in four very specific situations:
```

**Review by marsonya — CHANGES_REQUESTED** [review]

**Review by raisedadead — APPROVED** [review]

**Review by marsonya — APPROVED** [review]

**Comment by nhcarrigan** [comment]
@raisedadead before landing this, I asked Quincy to give it a once-over as well to make sure everything is still accurate and up-to-date.

**Comment by raisedadead** [comment]
I will let you merge this instead. Thanks for letting me know.

**Review by QuincyLarson — APPROVED** [review]
This all looks solid. No further edits recommended from me.

**Pull request merged**
